src/tests/integration/test_generic_db_operations.py

```diff
@@ -770,27 +753,140 @@ def test_sql_query(ip_with_dynamic_db, cell, request, test_table_name_dict):
     assert out.error_in_exec is None
 
 
-@pytest.mark.parametrize("ip_with_dynamic_db", ALL_DATABASES)
-def test_sql_query_cte_suggestion(ip_with_dynamic_db, request):
+@pytest.mark.parametrize(
+    "cell",
+    [
+        "%%sql\nSELECT * FROM subset",
+        "%%sql --with subset\nSELECT * FROM subset",
+    ],
+    ids=[
+        "cte-inferred",
+        "cte-explicit",
+    ],
+)
+@pytest.mark.parametrize(
+    "ip_with_dynamic_db",
+    [
+        "ip_with_postgreSQL",
+        "ip_with_mySQL",
+        "ip_with_mariaDB",
+        "ip_with_SQLite",
+        "ip_with_duckDB_native",
+        "ip_with_duckDB",
+        pytest.param(
+            "ip_with_MSSQL",
+            marks=pytest.mark.xfail(
+                reason="We need to close any pending results for this to work"
+            ),
+        ),
+        "ip_with_Snowflake",
+        "ip_with_oracle",
+    ],
+)
+def test_sql_query_cte(ip_with_dynamic_db, request, test_table_name_dict, cell):
```

Comment:
Seems like the test case is for testing whether the saved CTE is queryable,

But if we do `SELECT * FROM {test_table_name_dict['numbers']}`, is it directly querying to the table?

Reply:
this is storing the CTE:

```
    ip_with_dynamic_db.run_cell(
        "%%sql --save subset --no-execute \n"
        f"SELECT * FROM {test_table_name_dict['numbers']}"
    )
```

then, the next line (parameterized via `cell`) will query the subset CTE

Reply:
Cool! Thanks for confirmation